### DISCUSSION TITLE
Migrate - Operation not permitted

### DISCUSSION THREAD

**Discussion by jCaffie** [netbox-community/netbox#16369] [Help Wanted!]
### **Deployment Type**
Self-hosted

### **NetBox Version**
v4.0.3

### **Python Version**
3.12

### **Steps to Reproduce**
Attempt to upgrade from 3.7.8 to 4.0.3

### **Observed Behavior**

I am in an air gapped environment and my netbox server is running Rocky Linux 9.0.

Our server with internet access, I performed an upgrade from 3.7.8 to 4.0.3. I ran a successful upgrade.sh and pip downloaded the requirements.txt and local_requirements.txt packages, and SCP'd these files to my netbox server.

/opt/netbox is now using the 4.0.3 directory and I did a pip install and verified all of the packages were successfully installed for /opt/netbox/venv/bin/python3.12.
Also transferred necessary files from old version (local_requirements, ldap config, configuration.py).

### **TLDR**

When trying to run the manage.py migrate, I receive the following error:

```
(venv) [root@netboxserver netbox]# /opt/netbox/venv/bin/python3.12 /opt/netbox/netbox/manage.py migrate
/opt/netbox/venv/bin/python3.12: can't open file '/opt/netbox/netbox/manage.py': [Errno 1] Operation not permitted
```

I apologize in advance if this is a newbie linux problem.
I have tried everything i can think of and seen online such as disabling SELINUX, the permissions are all chmodded to 755, and im running as root.

Don't know what to do at this point so here I am.

**Comment by candlerb** [comment]
What does `dmesg | tail -10` show after running that command?

A permission error, when running as root, suggests a problem with SELinux or Apparmor (the former I believe, in the case of Rocky). You could try setting it into non-enforcing mode to confirm.

**Reply by jCaffie** [reply]
I mentioned in the post, I disabled SELINUX and still received the error.

But I did go ahead and try it again with SELINUX enforced and dmesg did not show any new logs.
